.circleci/config.yml

```diff
@@ -4,40 +4,46 @@ version: 2
   machine: true
   steps:
     - checkout
-    - run:
+    - run: &checkouttask
         name: Pull submodules
         command: |
           git submodule init
           git submodule update
+          git checkout $CIRCLECI_BRANCH
+          git reset --hard $CIRCLECI_SHA1
     - run:
         name: Install the emulation handlers
         command: docker run --rm --privileged multiarch/qemu-user-static:register --reset
     - run:
         name: Run the tests in a container
         command: |
-          docker run -e CI=true --rm -v `pwd`:`pwd` -w `pwd` "pqclean/ci-container:$ARCH" /bin/bash -c "
+          docker run -e CI=true -e PQCLEAN_ONLY_DIFF=1 --rm -v `pwd`:`pwd` -w `pwd` "pqclean/ci-container:$ARCH" /bin/bash -c "
           uname -a &&
           export CC=${CC} &&
           pip3 install -r requirements.txt &&
-          cd test && python3 -m nose --rednose --verbose"
+          mkdir test-results &&
+          cd test && python3 -m nose --rednose --verbose --with-xunit --xunit-file=../test-results/nosetests.xml"
+    - store_test_results:
+        path: test-results
```

Comment:
Can you explain to me what this is doing?

Reply:
It produces slightly nicer result output on the CircleCI pages.  See test summary on https://circleci.com/gh/PQClean/PQClean/3161